sorc/ncep_post.fd/CMakeLists.txt

```diff
@@ -0,0 +1,265 @@
+    MDL2STD_P.f
+    MDL2THANDPV.f
+    MDLFLD.f
+    MICROINIT.F
+    MISCLN.f
+    MIXLEN.f
+    MPI_FIRST.f
+    MPI_LAST.f
+    MSFPS.f
+    NGMFLD.f
+    NGMSLP.f
+    OTLFT.f
+    OTLIFT.f
+    PARAMR.f
+    params.F
+    PARA_RANGE.f
+    physcons.f
+    PMICRPH.f
+    POLEAVG.f
+    PROCESS.f
+    READCNTRL.F
+    READ_xml.f
+    RHGRD.f
+    RQSTFLD.F
+    SCLFLD.f
+    SELECT_CHANNELS.f
+    SERVER.f
+    SET_LVLSXML.f
+    SET_OUTFLDS.f
+    SETUP_SERVERS.f
+    SLP_new.f
+    SLP_NMM.f
+    SMOOTH.f
+    SNFRAC.f
+    SNFRAC_GFS.f
+    SOIL_mod.f
+    SPLINE.f
+    SURFCE.f
+    svptbl.f
+    TABLE.f
+    TABLEQ.f
+    TRPAUS.f
+    TTBLEX.f
+    VRBLS2D_mod.f
+    VRBLS3D_mod.f
+    VRBLS4D_mod.f
+    WETBULB.f
+    WETFRZLVL.f
+    xml_perl_data.f
+    ZENSUN.f)
+
+list(APPEND EXE_SRC
+    ASSIGNNEMSIOVAR.f
+    GETNEMSNDSCATTER.f
+    GFSPOSTSIG.F
+    INITPOST.F
+    INITPOST_GFS.f
+    INITPOST_GFS_NEMS.f
+    INITPOST_GFS_NEMS_MPIIO.f
+    INITPOST_GFS_NETCDF.f
+    INITPOST_GFS_NETCDF_PARA.f
+    INITPOST_GFS_SIGIO.f
+    INITPOST_NEMS.f
+    INITPOST_NEMS_MPIIO.f
+    INITPOST_NETCDF.f
+    INITPOST_NMM.f
+    TIMEF.f
+    WRFPOST.f
+    getIVariableN.f
+    getVariable.f
+    getlvls.f
+    intio_tags.f
+    native_endianness.f
+    retrieve_index.f
+    wrf_io_flags.f
+    wrf_io_flags.h)
+
+# Use IO stubs in place of WRFLIB
+if(NOT wrf_FOUND)
+  list(APPEND EXE_SRC
+    io_int_stubs.f)
+endif()
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS
+      "-g -traceback -fp-model source -free -convert big_endian")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS
+      "-g -fbacktrace -ffree-form -ffree-line-length-none -fconvert=big-endian")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check")
+endif()
+
+set(LIBNAME nceppost)
+set(EXENAME ncep_post)
+
+set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include")
+
+add_library(${LIBNAME} STATIC ${LIB_SRC})
+add_library(${PROJECT_NAME}::${LIBNAME} ALIAS ${LIBNAME})
+
+set_target_properties(${LIBNAME} PROPERTIES Fortran_MODULE_DIRECTORY
+                                            ${module_dir})
+
+target_include_directories(${LIBNAME} INTERFACE
+  $<BUILD_INTERFACE:${module_dir}>
+  $<INSTALL_INTERFACE:include>)
+
+target_link_libraries(${LIBNAME} PUBLIC
+  bacio::bacio_4
+  crtm::crtm
+  g2::g2_4
+  g2tmpl::g2tmpl
+  ip::ip_4
+  MPI::MPI_Fortran
+  NetCDF::NetCDF_Fortran)
+
+target_link_libraries(${LIBNAME} PRIVATE
+  sp::sp_4
+  w3emc::w3emc_4
+  w3nco::w3nco_4)
+
+if(OpenMP_Fortran_FOUND)
+  target_link_libraries(${LIBNAME} PUBLIC OpenMP::OpenMP_Fortran)
+endif()
+
+if(BUILD_POSTEXEC)
+  add_executable(${EXENAME} ${EXE_SRC})
+  target_link_libraries(${EXENAME} PRIVATE
```

Comment:
Do you not need to link to `sp`, `w3emc`, and `w3nco` on the executable since you made them private for the lib and it would be the `d` version? Or did I misunderstand some of that.

Reply:
``sp_4``, ``w3emc_4``, and ``w3nco_4`` will be linked to ``ncep_post`` executable as a private dependency of ``nceppost`` target but will not be passed as a dependency to users of this librray, which is what we want so that ufs model can have direct dependency on ``_d`` versions.